src/lib.rs

```diff
@@ -335,11 +347,11 @@ impl Version {
     /// Assume it's a cargo Version
     #[cfg(feature = "cargo-projects")]
     pub fn cargo(&self) -> &semver::Version {
-        #[allow(irrefutable_let_patterns)]
-        if let Version::Cargo(v) = self {
-            v
-        } else {
-            panic!("Version wasn't in the cargo format")
+        #[allow(unreachable_patterns)]
+        match self {
+            // For now treat generic versions as cargo versions
+            Version::Cargo(v) | Version::Generic(v) => v,
```

Comment:
We pretend generic versions are cargo versions; I'm not sure how exactly we want to handle non-Cargo versions longterm.